python/ray/workflow/http_event_provider.py

```diff
@@ -0,0 +1,227 @@
+                pass
+            else:
+                raise WorkflowEventHandleError(workflow_id, f"The same {event_key} is used to get payload again.")
+        self.event_key_payload[workflow_id][event_key] = asyncio.Future()
+        return await self.event_key_payload[workflow_id][event_key]
+
+    async def report_checkpointed(self, event_key, confirmation) -> str:
+        """Internal method used by HTTPListner to confirm the received event has been checkpointed by workflow
+        """
+        try:
+            self.event_checkpoint_pending[event_key].set_result(confirmation)
+        except KeyError:
+            return f"{event_key} cannot be found to acknowledge request. " \
+                   f"The event provider may have been restarted."
+        return "OK"
+
+
+
+class HTTPListener(EventListener):
+    """HTTPLister is defined to work with the HTTPEventProvider. It implements two
+    APIs, poll_for_event() and event_checkpointed(). An instance of HTTPListener will
+    be started by the get_message() of the workflow.wait_for_event() to listen for an event
+    from the HTTPEventProvider instance (a Ray Serve deployment). Another instance of HTTPListener
+    will be started by the message_committed() of the workflow.wait_for_event() to confirm
+    that the event has been checkpointed.
+
+
+    Example definition
+    ==================
+
+    ```
+    class HTTPListener:
+
+        def __init__(self):
+            self.handle = ray.serve.get_deployment().get_handle()
+
+        async def poll_for_event(self, *args, **kwargs) -> Any:
+
+        async def event_checkpointed(self, event) -> None:
+
+    ```
+
+    Example Usage
+    =============
+    >>> from ray.workflow.http_event_provider import HTTPEventProvider, HTTPListener
+    >>> ray.init(address='auto', namespace='serve')
+    >>> serve.start(detached=True)
+    >>> HTTPEventProvider.deploy()
+    >>> event_node = workflow.wait_for_event( # doctest: +SKIP
+    ...     HTTPListener, workflow_id='', event_key='')
+    >>> handle_event = ... # doctest: +SKIP
+    >>> handle_event.bind(event_node).run() # doctest: +SKIP
+    >>>
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.handle = ray.serve.get_deployment('WorkflowEventProvider').get_handle(sync=True)
+
+    async def poll_for_event(self, *args, **kwargs) -> Event:
+        """workflow.wait_for_event calls this method to subscribe to the HTTPEventProvider
+        and return the received external event
+        Args:
+            kwargs: event_key, a unique identifier to the receiving node in a workflow
+        Returns:
+            tuple(event_key, event_payload)
+        """
+        event_key = kwargs.pop("event_key", None)
+        workflow_id = workflow_context.get_current_workflow_id()
+        if event_key is None:
+            raise WorkflowEventHandleError(workflow_id, \
+                                           "poll_for_event() needs kwargs: event_key=''")
```

Comment:
Using the right signature and the configure is not necessary.

Reply:
Also let's just use step id as the default one.

Reply:
Done